### PR TITLE
Add IfcGeoScienceElement and subtypes

### LINKED ISSUE #393: [IFC Tunnel] IfcGeoScienceModel and IfcGeoScienceFeature
Updated UML model:
image

image

_Originally posted by @larswik in https://github.com/bSI-InfraRoom/IFC-Specification/issues/315#issuecomment-1060458394_

### PR DESCRIPTION
**Pull request by larswik**
Fixes #393

### CODE REVIEW

File: IFC4x3/Constants/d/DESTRUCTIVEDRILLING_1jCV42H110velcrgc01CVe.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="DESTRUCTIVEDRILLING_1jCV42H110velcrgc01CVe" Name="DESTRUCTIVEDRILLING" UniqueId="6d31f102-4410-40e6-8be6-d6a98004c7e8">
+	<Documentation>A drilling process in which the boring is performed using destructive tools</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/d/DESTRUCTIVEDRILLING_1jCV42H110velcrgc01CVe.xml]:
```suggestion
	<Documentation>A drilling process in which the boring is performed using destructive tools.</Documentation>
```

File: IFC4x3/Constants/f/FAULT_2QdZLdMmj6pf3tQXnP_Cwb.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FAULT_2QdZLdMmj6pf3tQXnP_Cwb" Name="FAULT" UniqueId="9a9e3567-5b0b-46ce-90f7-6a1c59f8cea5">
+	<Documentation>A shear displacement structure includes all brittle to ductile style structures along which displacement has occurred, from a simple, single &apos;planar&apos; brittle or ductile surface to a fault system comprised of tens of strands of both brittle and ductile nature. This structure may have some significant thickness (a deformation zone) and have an associated body of deformed rock that may be considered a deformation unit (which geologicUnitType is ‘DeformationUnit’) which can be associated to the ShearDisplacementStructure using GeologicFeatureRelation from the GeoSciML Extension package</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/f/FAULT_2QdZLdMmj6pf3tQXnP_Cwb.xml]: What are all the references to various things (some attributes, XMLs etc.)? Would make sense to have these properly documented since we are working on ISO standard documentation.
```suggestion
	<Documentation>A shear displacement structure includes all brittle to ductile style structures along which displacement has occurred, from a simple, single &apos;planar&apos; surface to a fault system comprised of tens of strands of both brittle and ductile nature. This structure may have some significant thickness (a deformation zone) and have an associated body of deformed rock that may be considered a deformation unit (which geologicUnitType is ‘DeformationUnit’) which can be associated to the ShearDisplacementStructure using GeologicFeatureRelation from the GeoSciML Extension package</Documentation>
```

File: IFC4x3/Constants/f/FOLD_1d9W6iy25Fp8VZfJLka_AM.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FOLD_1d9W6iy25Fp8VZfJLka_AM" Name="FOLD" UniqueId="672601ac-f021-4fcc-87e3-a5356e93e296">
+	<Documentation>A fold is formed by one or more systematically curved layers, surfaces, or lines in a rock body. A fold denotes a structure formed by the deformation of a geologic structure, such as a contact which the original undeformed geometry is presumed, to form a structure that may be described by the translation of an abstract line (the fold axis) parallel to itself along some curvilinear path (the fold profile). Folds have a hinge zone (zone of maximum curvature along the surface) and limbs (parts of the deformed surface not in the hinge zone). Folds are described by an axial surface, hinge line, profile geometry, the solid angle between the limbs, and the relationships between adjacent folded surfaces if the folded structure is a Layering fabric.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/f/FOLD_1d9W6iy25Fp8VZfJLka_AM.xml]:
```suggestion
	<Documentation>A fold is formed by one or more systematically curved layers, surfaces, or lines in a rock body. A fold denotes a structure formed by the deformation of a geologic structure, such as a contact of which the original undeformed geometry is presumed, to form a structure that may be described by the translation of an abstract line (the fold axis) parallel to itself along some curvilinear path (the fold profile). Folds have a hinge zone (zone of maximum curvature along the surface) and limbs (parts of the deformed surface not in the hinge zone). Folds are described by an axial surface, hinge line, profile geometry, the solid angle between the limbs, and the relationships between adjacent folded surfaces if the folded structure is a layering fabric.</Documentation>
```

File: IFC4x3/Constants/g/GEOLOGICUNIT_2gVB7PHR1DjRyfPvxAeXQ9.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="GEOLOGICUNIT_2gVB7PHR1DjRyfPvxAeXQ9" Name="GEOLOGICUNIT" UniqueId="aa7cb1d9-45b0-4db5-bf29-679ecaa21689">
+	<Documentation>A GeologicUnit may represent a body of material in the Earth whose complete and precise extent is inferred to exist (e.g., North American Data Model GeologicUnit, Stratigraphic unit in the sense of NACSN, or International Stratigraphic Code ), or a classifier used to characterize parts of the Earth (e.g. lithologic map unit like &apos;granitic rock&apos; or &apos;alluvial deposit&apos;, surficial units like &apos;till&apos; or &apos;old alluvium&apos;). It includes both formal units (i.e. formally adopted and named in an official lexicon) and informal units (i.e. named but not promoted to a lexicon) and unnamed units (i.e., recognizable, described and delineable in the field but not otherwise formalised). In simpler terms, a geologic unit is a package of earth material (generally rock or soil).</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/g/GEOLOGICUNIT_2gVB7PHR1DjRyfPvxAeXQ9.xml]:
```suggestion
	<Documentation>May represent a body of material in the Earth whose complete and precise extent is inferred to exist (e.g., North American Data Model GeologicUnit, Stratigraphic unit in the sense of NACSN, or International Stratigraphic Code ), or a classifier used to characterize parts of the Earth (e.g. lithologic map unit like &apos;granitic rock&apos; or &apos;alluvial deposit&apos;, surficial units like &apos;till&apos; or &apos;old alluvium&apos;). It includes both formal units (i.e. formally adopted and named in an official lexicon) and informal units (i.e. named but not promoted to a lexicon) and unnamed units (i.e., recognizable, described and delineable in the field but not otherwise formalised). In simpler terms, a geologic unit is a package of earth material (generally rock or soil).</Documentation>
```

File: IFC4x3/Constants/g/GEOLOGYMODEL_1SEPegtojBEQLkFPqms$01.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="GEOLOGYMODEL_1SEPegtojBEQLkFPqms01" Name="GEOLOGYMODEL" UniqueId="5c399a2a-df2b-4b39-a56e-3d9d30dbf001">
+	<Documentation>Model of geological structured as considered relevant for the project, as a base for the definition of building- and design-related geotechnical models, hydrogeological models and GeoHazardModel</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/g/GEOLOGYMODEL_1SEPegtojBEQLkFPqms$01.xml]:
```suggestion
	<Documentation>Model of geological structure as considered relevant for the project, as a base for the definition of building- and design-related geotechnical models, hydrogeological models and GeoHazardModel</Documentation>
```

File: IFC4x3/Constants/g/GEOTECHNICALUNIT_0LCxv_KUzEd9bwJfHXQPlw.xml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="GEOTECHNICALUNIT_0LCxv_KUzEd9bwJfHXQPlw" Name="GEOTECHNICALUNIT" UniqueId="1533be7e-51ef-4e9c-997a-4e9461699bfa">
+	<Documentation>A surface or a volume in which the mechanical behaviour and other design-relevant characteristics are characterized using the same geotechnical parameters values. Several alternative classifications (=GeotechModels) can be required in a project for different design tasks.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/g/GEOTECHNICALUNIT_0LCxv_KUzEd9bwJfHXQPlw.xml:3]: "(=GeotechModels)" this is a confusing notation. What does it mean?

File: IFC4x3/Constants/h/HAZARDAREA_3CPujkgkb5rfeoDTJkPNVx.xml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HAZARDAREA_3CPujkgkb5rfeoDTJkPNVx" Name="HAZARDAREA" UniqueId="cc678b6e-aae9-45d6-9a32-35d4ee6577fb">
+	<Documentation>Discrete spatial objects representing a natural hazard.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/h/HAZARDAREA_3CPujkgkb5rfeoDTJkPNVx.xml]:
```suggestion
	<Documentation>Discrete spatial object representing a natural hazard.</Documentation>
```

File: IFC4x3/Constants/h/HYDROGEOMODEL_3GnJynP9vAVAIV$zNyW$1R.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HYDROGEOMODEL_3GnJynP9vAVAIVzNyW1R" Name="HYDROGEOMODEL" UniqueId="d0c53f31-649e-4a7c-a49f-ffd5fc83f05b">
+	<Documentation>Model of hydrogeological conditions with parameters like permeability and definition of boundary conditions for hydrogeological modelling</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/h/HYDROGEOMODEL_3GnJynP9vAVAIV$zNyW$1R.xml]:
```suggestion
	<Documentation>Model of hydrogeological conditions with parameters like permeability and definition of boundary conditions for hydrogeological modelling.</Documentation>
```

File: IFC4x3/Constants/h/HYDROGEOUNIT_1szVAsDb91gR0fVuZJxcN9.xml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HYDROGEOUNIT_1szVAsDb91gR0fVuZJxcN9" Name="HYDROGEOUNIT" UniqueId="76f5f2b6-3652-41a9-b029-7f88d3ee65c9">
+	<Documentation>Any soil or rock unit or zone that by virtue of its hydraulic properties has a distinct influence on the storage or movement of groundwater (after ANS, 1980).</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/h/HYDROGEOUNIT_1szVAsDb91gR0fVuZJxcN9.xml:3]: Reference does not seem according to IFC documentation.

File: IFC4x3/Constants/p/PHYSICALPROPERTYDISTRIBUTION_0JNVWhdPzDpwj69EkBY6Sf.xml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PHYSICALPROPERTYDISTRIBUTION_0JNVWhdPzDpwj69EkBY6Sf" Name="PHYSICALPROPERTYDISTRIBUTION" UniqueId="135df82b-9d9f-4dcf-ab46-24eb8b886729">
+	<Documentation>Additional option (alternative to discrete models) to describe a +/- continuous spatial distribution of any physical parameter (geotechnical key-parameters, permeabilty, likelyhood of. e.g. a fault or any other uncertainty-related information)</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/p/PHYSICALPROPERTYDISTRIBUTION_0JNVWhdPzDpwj69EkBY6Sf.xml:3]: Are the constants here the same? Could remove one?

File: IFC4x3/Constants/u/USERDEFINED_30DlT5Cz5CEOn$eGTBzZaB.xml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="USERDEFINED_30DlT5Cz5CEOneGTBzZaB" Name="USERDEFINED" UniqueId="c036f745-33d1-4c39-8c7f-a1074bf6390b">
```
Comment (SergejMuhic) [inline IFC4x3/Constants/u/USERDEFINED_30DlT5Cz5CEOn$eGTBzZaB.xml]: You could re-use USERDEFINED and NOTDEFINED.

File: IFC4x3/Constants/v/VOIDBODY_35pQibG99EYh1mNDyAZ1QI.xml [hunk 12]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VOIDBODY_35pQibG99EYh1mNDyAZ1QI" Name="VOIDBODY" UniqueId="c5cdab25-4092-4e8a-b070-5cdf0a8c1692">
+	<Documentation>A discrete air filled geological feature, including caves and other voids</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Constants/v/VOIDBODY_35pQibG99EYh1mNDyAZ1QI.xml]:
```suggestion
	<Documentation>A discrete air filled geological feature, including caves and other voids.</Documentation>
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcBorehole/Documentation.md [hunk 13]
```diff
@@ -1,2 +1,4 @@
 Representation of the concept of a linear geological and geotechnical model, usually an interpretation but sometimes created direct from ground penetrating measurement  
 The assembly may contain one of more strata and other elements such as capping and lining. The contained subtypes of _IfcGeotechnicalStratum_ will have shape representations made from straight or bent tubes reflecting the bore diameter, or discs if a 'Yabuki' top surface model is being used.
+
+NOTE Definition from OGC GeoSciML: A Borehole is the generalized term for any narrow shaft drilled in the ground, either vertically, horizontally, or inclined.
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcBorehole/Documentation.md]:
```suggestion
>NOTE Definition from OGC GeoSciML: A Borehole is the generalized term for any narrow shaft drilled in the ground, either vertically, horizontally, or inclined.
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceElement/DocEntity.xml [hunk 14]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeoScienceElement" UniqueId="c765fabf-302f-4b26-b2bc-939f814c6eae" BaseDefinition="IfcElement" />
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceElement/DocEntity.xml:2]: No attributes? Should it be abstract?

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceFeature/DocEntity.xml [hunk 15]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeoScienceFeature" UniqueId="179a94fb-7e68-4d91-89ad-ac03ff8693f4" BaseDefinition="IfcGeoScienceElement" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="124ad9f3-ef29-4e12-a9b5-8e9cb76959a7" DefinedType="IfcGeoScienceFeatureTypeEnum" AttributeFlags="1">
+			<Documentation>Identifies the predefined type of geoscience feaures. This type may associate additional specific property sets.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceFeature/DocEntity.xml]:
```suggestion
			<Documentation>Identifies the predefined type of geoscience features. This type may associate additional specific property sets.</Documentation>
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceModel/Documentation.md [hunk 16]
```diff
@@ -0,0 +1 @@
+Model of geological structured as considered relevant for the project, as a base for the definition of building- and design-related geotechnical models, hydrogeological models and GeoHazardModel
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceModel/Documentation.md]: Assumption.
```suggestion
Model of the geological structure as considered relevant for the project, as a base for the definition of building- and design-related geotechnical models, hydrogeological models and geo hazard model.
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcBoreholeTypeEnum/DocEnumeration.xml [hunk 17]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcBoreholeTypeEnum" UniqueId="f73f0cd4-537d-4d62-8137-14f0bb74e802">
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcBoreholeTypeEnum/DocEnumeration.xml:2]: Any documentation for the enum?

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceFeatureTypeEnum/DocEnumeration.xml [hunk 18]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeoScienceFeatureTypeEnum" UniqueId="b42aac0e-e587-414f-af6d-5e33d8915603">
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceFeatureTypeEnum/DocEnumeration.xml:2]: Documentation?

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceModelTypeEnum/DocEnumeration.xml [hunk 19]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeoScienceModelTypeEnum" UniqueId="237901cb-201a-430f-b80d-874afd89f32a">
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceModelTypeEnum/DocEnumeration.xml:2]: Documentation?

**Review by SergejMuhic — CHANGES_REQUESTED** [review]
The WRs I will just try to compile later. Have not checked them.

**Review by anborr — APPROVED** [review]

File: IFC4x3/Constants/u/USERDEFINED_30DlT5Cz5CEOn$eGTBzZaB.xml (hunk 11 above)
Comment (larswik) [reply IFC4x3/Constants/u/USERDEFINED_30DlT5Cz5CEOn$eGTBzZaB.xml]: Did that now.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceModelTypeEnum/DocEnumeration.xml (hunk 19 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceModelTypeEnum/DocEnumeration.xml:2]: Done (see below).

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceFeatureTypeEnum/DocEnumeration.xml (hunk 18 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceFeatureTypeEnum/DocEnumeration.xml:2]: Done (see below).

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcBoreholeTypeEnum/DocEnumeration.xml (hunk 17 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcBoreholeTypeEnum/DocEnumeration.xml:2]: Done (see below).

**Comment by larswik** [comment]
I saw that the same change is needed here as at many other places since there is no corresponding type object:
- Make PredefinedType mandatory
- Change WR to HasObjectType

**Comment by larswik** [comment]
> I saw that the same change is needed here as at many other places since there is no corresponding type object:
> 
> * Make PredefinedType mandatory
> * Change WR to HasObjectType

Done.

File: IFC4x3/Constants/g/GEOTECHNICALUNIT_0LCxv_KUzEd9bwJfHXQPlw.xml (hunk 6 above)
Comment (larswik) [reply IFC4x3/Constants/g/GEOTECHNICALUNIT_0LCxv_KUzEd9bwJfHXQPlw.xml:3]: Yep, I see,,,I think the whole sentence is difficult to understand actually. I think that the domain team must be asked about this one. I think that what they mean is that a project may require several _IfcGeoScienceModel_ instances where the contained _GEOTECHNICALUNIT_ elements describe different aspects of the geotechnical situation.

File: IFC4x3/Constants/p/PHYSICALPROPERTYDISTRIBUTION_0JNVWhdPzDpwj69EkBY6Sf.xml (hunk 10 above)
Comment (larswik) [reply IFC4x3/Constants/p/PHYSICALPROPERTYDISTRIBUTION_0JNVWhdPzDpwj69EkBY6Sf.xml:3]: Yes. My mistake obviously.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceElement/DocEntity.xml (hunk 14 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceElement/DocEntity.xml:2]: Yes!

File: IFC4x3/Constants/h/HYDROGEOUNIT_1szVAsDb91gR0fVuZJxcN9.xml (hunk 9 above)
Comment (larswik) [reply IFC4x3/Constants/h/HYDROGEOUNIT_1szVAsDb91gR0fVuZJxcN9.xml:3]: I can't be sure what "ANS, 1980" actually is? Have to ask domain team.

**Comment by SergejMuhic** [comment]
Let me know if you want to merge this one. Of course after extracting the unresolved comments into issues.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcBorehole/Documentation.md [hunk 20]
```diff
@@ -1,2 +1,4 @@
 Representation of the concept of a linear geological and geotechnical model, usually an interpretation but sometimes created direct from ground penetrating measurement
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcBorehole/Documentation.md:1]:
```suggestion
Representation of a linear geological and geotechnical model concept, usually an interpretation but sometimes created direct from ground penetrating measurement  
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceFeature/DocEntity.xml [hunk 21]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeoScienceFeature" UniqueId="179a94fb-7e68-4d91-89ad-ac03ff8693f4" BaseDefinition="IfcGeoScienceElement" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="124ad9f3-ef29-4e12-a9b5-8e9cb76959a7" DefinedType="IfcGeoScienceFeatureTypeEnum">
+			<Documentation>Identifies the predefined type of geoscience features. This type may associate additional specific property sets.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceFeature/DocEntity.xml]:
```suggestion
			<Documentation>Specifies the type for which the value is selected from a predefined type enumeration. This type may associate additional specific property sets.</Documentation>
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceModel/DocEntity.xml [hunk 22]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeoScienceModel" UniqueId="444b3163-ad13-47a0-b903-7c8955684265" BaseDefinition="IfcGeoScienceElement" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="82a92e9d-9ceb-457f-9068-9e706c650490" DefinedType="IfcGeoScienceModelTypeEnum">
+			<Documentation>Identifies the predefined type of geoscience models. This type may associate additional specific property sets.</Documentation>
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceModel/DocEntity.xml]:
```suggestion
			<Documentation>Specifies the type for which the value is selected from a predefined type enumeration. This type may associate additional specific property sets.</Documentation>
```

**Review by SergejMuhic — APPROVED** [review]
Harmonizing PDT descriptions.

**Review by SergejMuhic — APPROVED** [review]
